src/applications/disability-benefits/view-payments/components/view-payments-lists/payments/Payments.jsx

```diff
@@ -17,17 +17,73 @@ class PaymentsReceived extends Component {
     paginatedData: null,
     fromNumber: null,
     toNumber: null,
+    currentSortOrder: 'ASC',
+    nextSortOrder: 'DESC',
+    sortingOn: null,
   };
 
   componentDidMount() {
     this.handleLoadData();
   }
 
+  dynamicSort = (property, sort) => {
+    let sortOrder = 1;
+    if (property[0] === '-') {
+      sortOrder = -1;
+      // eslint-disable-next-line no-param-reassign
+      property = property.substr(1);
+    }
+    if (sort === 'ASC') {
+      return function(a, b) {
+        const result =
+          // eslint-disable-next-line no-nested-ternary
+          a[property] < b[property] ? -1 : a[property] > b[property] ? 1 : 0;
+        return result * sortOrder;
+      };
+    } else {
+      return function(a, b) {
+        const result =
+          // eslint-disable-next-line no-nested-ternary
+          a[property] > b[property] ? -1 : a[property] < b[property] ? 1 : 0;
+        return result * sortOrder;
```

Comment:
If these functions are sorting numbers, which I'm guessing since the file name is "Payments", it's usually better/easier to use this method:

```
result = (b[property] - a[property]) * sortOrder;
```

Although you might need to parse the number first.

Reply:
the current logic works for both numbers and strings. what would be the benefit of using your method?

Reply:
You'll get the numbers to sort based on the string value and not the numeric value - see https://jsbin.com/jividatuqi/edit?js,console

```js
'1,2,3,4,5,6,7,8,9,10,11,12'.split(',').sort((a, b) => {
  if (a === b) return 0;
  return a > b ? 1 : -1;
});
// result = ["1", "10", "11", "12", "2", "3", "4", "5", "6", "7", "8", "9"]

Reply:
So you'll actually need to differentiate numeric and alpha(numeric) columns.

Reply:
my understanding is that in practice this is actually fine since the character codes for numbers as strings as both sequential and come before strings

Reply:
Yes, but if you're sorting payment numbers, users would expect a numerical sort. What do your designers say about this?

Reply:
what I mean is If you are sorting a numerical column they will sort according to their ascii value, which is still sequential. We will not have columns for this table that are mixed with some rows numerical and some strings, all columns that are strings are only strings and all columns that are numerical are only numberical.

Reply:
If you look at the example I shared... `1, 10, 11, 12, 2` is how the string of numbers would sort. If you keep the current code, but parse numerical values, it should properly sort the numbers.

Reply:
@Mottie have you tested my branch? It sorts numeric values just fine for me and @micahchiang

Reply:
@Mottie Ah, I see what you mean. Let me fiddle with this a bit.